src/cleanup_queue-inl.h

```diff
@@ -29,6 +31,18 @@ void CleanupQueue::Remove(Callback cb, void* arg) {
   cleanup_hooks_.erase(search);
 }
 
+constexpr std::strong_ordering CleanupQueue::CleanupHookCallback::operator<=>(
```

Comment:
Does it make our code faster/better somehow? I wasn't aware of this overload with std::strong_ordering and reading it at first glace seems an scary complexity to our code

Reply:
I might be biased (**coughs** since I authored it) but this change makes sorting extremely easy (and readable). std::ranges and std::views have lots of uncharted beauties such as sort. I haven't benchmarked but if requested(or someone has time to so it) we can do it.

Reply:
No worries. It might be just because it's new for me. Let's keep it.